### ISSUE TITLE
Rule Change: (valid-typeof) undefined should be banned

### ISSUE THREAD

**Issue by Zzzen** [eslint/eslint#15591]
### What rule do you want to change?

valid-typeof

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
typeof foo === null
typeof foo === undefined
```

### What does the rule currently do for this code?

only `null` is reported

### What will the rule do after it's changed?

both `null` and `undefined` should be reported

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
This was partially discussed in https://github.com/eslint/eslint/issues/14074.

I think it makes sense to report global `undefined` regardless of `requireStringLiterals`option. It's a small change in the functionality of this rule that would catch probably the most common mistakes with `typeof`.

I'll champion this.

(As a future enhancement and a much bigger change in the functionality, this rule could use [getStaticValue](https://eslint-utils.mysticatea.dev/api/ast-utils.html#getstaticvalue). That would catch global `undefined` since it evaluates to the `undefined` value, and more).

**Comment by nzakas** [comment]
This change makes sense.
